File: Medium/406.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
  public int[][] ReconstructQueue(int[][] people) {
    Array.Sort(people, (a, b) => {
      if (a[0] == b[0]) {
        return a[1].CompareTo(b[1]); // k 升序
      }

      return b[0].CompareTo(a[0]); // 身高降序
    });

    List<int[]> output = new List<int[]>();

    foreach (int[] p in people) {
      output.Insert(p[1], p);
    }

    return output.ToArray();
  }
}

File: Medium/406.py
from typing import List


class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        # 先看 -height：身高越高，-height 越小，所以排前面 再看 k：如果身高一样，k 越小越靠前
        people.sort(key=lambda p: (-p[0], p[1]))

        output = []

        for p in people:
            # 在 p[1] 这个 index 插入 p
            output.insert(p[1], p)

        return output